cookbook/core/control_flow/dynamics.py

```diff
@@ -1,56 +1,122 @@
+# Therefore, ``@dynamic`` decorator has to be used.
+#
+# Dynamic workflow is effectively both a task and a workflow. The key thing to note is that the ``body of tasks is run at run time and the
+# body of workflows is run at compile (aka registration) time``. Essentially, this is what a dynamic workflow leverages -- it’s a workflow that is compiled at run time (the best of both worlds)!
 #
-# Local execution will also work as it's treated like a task that will be run with Python native inputs.
+# At execution (run) time, Flytekit runs the compilation step, and produces
+# a ``WorkflowTemplate`` (from the dynamic workflow), which Flytekit then passes back to Flyte Propeller for further running, much like how sub-workflows are handled.
 #
-# The name of this function was intentionally chosen. Even though these are implemented as tasks in flytekit, you
-# should think about them as workflows. At execution time, flytekit is running the compilation step, and producing
-# a ``WorkflowTemplate``, which it then passes back to Flyte Propeller for further running, much like how sub-workflows
-# are handled.
+# We now define the dynamic workflow encapsulating the above mentioned points.
 @dynamic
-def my_subwf(a: int) -> (typing.List[str], int):
-    s = []
-    for i in range(a):
-        s.append(t1(a=i))
-    return s, 5
+def count_characters(s1: str, s2: str) -> int:
+    """
+    Calls the required tasks and returns the final result"""
 
+    # s1 and s2 are accessible
 
+    # initiliazing an empty list consisting of 26 empty slots corresponding to every alphabet (lower and upper case)
+    freq1 = [0] * 26
+    freq2 = [0] * 26
+
+    # looping through the string s1
+    for i in range(len(s1)):
+
+        # index and freq1 are not accesible as they are promises
+        index = return_index(character=s1[i])
+        freq1 = update_list(freq_list=freq1, list_index=index)
+
+    # looping through the string s2
+    for i in range(len(s2)):
+
+        # index and freq2 are not accesible as they are promises
+        index = return_index(character=s2[i])
+        freq2 = update_list(freq_list=freq2, list_index=index)
+
+    # counting the common characters
+    return derive_count(freq1=freq1, freq2=freq2)
+
+
+# %%
+# When tasks are called within any workflow, they return Promise objects. Likewise, in a dynamic workflow, the tasks' outputs are Promise objects that cannot be directly accessed (they shall be fulfilled by Flyte later).
+# Because of this fact, operations on the ``index`` variable like ``index + 1`` are not valid.
+# To manage this problem, the values need to be passed to the other tasks to unwrap them.
+#
+# A point to note is that local execution will work when a ``@dynamic`` decorator is used because Flytekit treats it like a ``task`` that will run with the Python native inputs. 
```

Comment:
Instead of saying "A point to note" can we use more rst syntax?  You can call out a note with `.. note::` See https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#boxes

Reply:
Done.